sdk/core/Azure.Core/src/HttpMessage.cs

```diff
@@ -12,95 +13,83 @@ namespace Azure.Core
     /// <summary>
     /// Represents a context flowing through the <see cref="HttpPipeline"/>.
     /// </summary>
-    public sealed class HttpMessage : IDisposable
+    public sealed class HttpMessage : PipelineMessage
```

Comment:
We use inheritance where we want types to be substitutable for one another.  In addition, wrappers add a small amount of perf overhead.  I agree that since message types aren't end-user facing, inheritance is less important, but I don't see a problem with this one, and it makes it easier to reuse the functionality across the libraries.  Do you have a concern with this inheritance pattern, or was this just an observation/suggestion?

Reply:
An additional comment here is that inheritance is important to the message type so that we can write policies that operate on the same message/request/response, without having to duplicate a policy's processing logic.